frontend/src/features/LLM/MwasHypothesisGenerator.tsx

```diff
@@ -0,0 +1,129 @@
+import React from 'react';
+
+import { useLazyGetHypothesisQuery } from '../../api/client.ts';
+import { load } from '@fingerprintjs/botd';
+
+import IconButton from '@mui/material/IconButton';
+import BlurOnIcon from '@mui/icons-material/BlurOn';
+import Box from '@mui/material/Box';
+import Skeleton from '@mui/material/Skeleton';
+import Typography from '@mui/material/Typography';
+import Link from '@mui/material/Link';
+import Tooltip from '@mui/material/Tooltip';
+
+const MwasHypothesisGenerator = ({ identifiers, virusFamilies }) => {
+    const [getHypothesisText, { data: hypothesisData, isLoading: isLoadingHypothesis, error: errorHypothesis }] =
+        useLazyGetHypothesisQuery();
+
+    const onButtonClick = async () => {
+        try {
+            const botd = await load();
+            const detect = await botd.detect();
+            if (!detect || detect.bot === true) {
+                throw new Error('Bot detected');
+            }
+            getHypothesisText({
+                idColumn: 'bioproject',
+                ids: identifiers ? identifiers['bioproject'].single : [],
+                idRanges: identifiers ? identifiers['bioproject'].range : [],
+                virusFamilies: virusFamilies ? virusFamilies : [],
+                pageStart: 0,
+                pageEnd: 1000,
+                identifiers: identifiers,
+            });
+        } catch (error) {
+            console.error(error);
+        }
+    };
+    console.log(hypothesisData);
+    const renderPlaceholder = () => {
+        if (isLoadingHypothesis) {
+            return <Skeleton variant='text' width={'100%'} height={60} />;
+        }
+        return null;
+    };
+    const hypothesisTextIsNonEmpty = () => hypothesisData && hypothesisData?.text?.length > 0;
+
+    const processHypothesisText = (text) => {
+        const combinedRegex = /PRJ\w+|\|([^|]+)\|/g;
+        const combinedMatches = text.match(combinedRegex);
+        if (combinedMatches) {
+            const finalComponents: (string | JSX.Element)[] = [];
+            let lastIndex = 0;
+            combinedMatches.forEach((match, index) => {
+                const matchIndex = text.indexOf(match, lastIndex);
+                if (matchIndex > lastIndex) {
+                    finalComponents.push(text.substring(lastIndex, matchIndex));
+                }
+                if (/PRJ\w+/.test(match)) {
```

Comment:
some of this code is used in multiple places now, we should try [DRY](https://en.wikipedia.org/wiki/Don%27t_repeat_yourself) simple utils like this. Formatting helpers  can be moved to `frontend/src/common/utils/textFormatting.ts` or a new file under `features/LLM/textFormatiing.ts`

We'll have a similar setup for multiple LLM actions so it makes sense to setup the code for that now

Reply:
Moved code to features/LLM/textFormatting.tsx, see latest commit: d56d4bbb08682c5672d2ba3d502e6597fc476ab6